app/scenes/mfa/mfa.js

```diff
@@ -0,0 +1,111 @@
+        password: React.PropTypes.string.isRequired,
+        loginRequest: React.PropTypes.object.isRequired
+    };
+
+    componentWillMount() {
+        this.setState({
+            token: '',
+            error: null
+        });
+    }
+
+    componentWillReceiveProps(nextProps) {
+        // In case the login is successful the previous scene (login) will take care of the transition
+        if (this.props.loginRequest.status === RequestStatus.STARTED &&
+            nextProps.loginRequest.status === RequestStatus.FAILURE) {
+            this.props.actions.goBack();
+        }
+    }
+
+    handleInput = (token) => {
+        this.setState({
+            token,
+            error: null
+        });
+    };
+
+    submit = () => {
+        if (!this.state.token) {
+            this.setState({
+                error: this.props.intl.formatMessage({
+                    id: 'login_mfa.tokenReq',
+                    defaultMessage: 'Please enter an MFA token'
+                })
+            });
+            return;
+        }
+
+        this.props.actions.login(this.props.loginId, this.props.password, this.state.token);
+    };
+
+    render() {
+        const {formatMessage} = this.props.intl;
+
+        return (
+            <KeyboardAvoidingView
+                behavior='padding'
+                style={[GlobalStyles.container, GlobalStyles.signupContainer]}
+            >
+                <Image
+                    source={logo}
+                />
+                <FormattedText
+                    style={[GlobalStyles.header, GlobalStyles.label]}
+                    id='login_mfa.enterToken'
+                    defaultMessage="To complete the sign in process, please enter a token from your smartphone's authenticator"
+                />
+                <ErrorText error={this.state.error}/>
+                <TextInput
+                    value={this.state.token}
+                    onChangeText={this.handleInput}
+                    onSubmitEditing={this.submit}
+                    style={GlobalStyles.inputBox}
+                    autoCapitalize='none'
+                    autoCorrect={false}
+                    keyboardType='numeric'
+                    placeholder={formatMessage({id: 'login_mfa.token', defaultMessage: 'MFA Token'})}
+                    returnKeyType='go'
+                    underlineColorAndroid='transparent'
+                />
+                <Button
+                    onPress={this.submit}
+                    loading={false}
+                    containerStyle={GlobalStyles.signupButton}
+                >
+                    <FormattedText
+                        style={GlobalStyles.signupButtonText}
+                        id='mobile.components.select_server_view.proceed'
+                        defaultMessage='Proceed'
+                    />
+                </Button>
+                <KeyboardAvoidingView style={GlobalStyles.pagePush}/>
```

Comment:
Actually I thought the same thing but it was there in the other scenes, I can go and remove it from everywhere